stumpy/core.py

```diff
@@ -4210,3 +4206,70 @@ def _mpdist(
     MPdist = _select_P_ABBA_value(P_ABBA, k, custom_func)
 
     return MPdist
+
+
+def process_isconstant(T, m, T_subseq_isconstant, T_subseq_isfinite=None):
+    """
+    Compute the rolling isconstant for 1-D and 2-D arrays.
+    This is accomplished by comparing the min and max within each window and
+    assigning `True` when the min and max are equal and `False` otherwise. If
+    a subsequence contains at least one NaN, then the subsequence is not constant.
+
+    If `T_subseq_isconstant` is provided as boolean array, its element will be set
+    to False if their corresponding value in `T_subseq_isfinite` is False.
+
+    Parameters
+    ----------
+    T : numpy.ndarray
+        The input 1D or 2D array
+
+    m : numpy.ndarray
+        The rolling window size
+
+    T_subseq_isconstant : np.ndarray, function, or list, default None
+        A parameter that is used to show whether a subsequence of a time series in `T`
+        is constant (True) or not. T_subseq_isconstant can be a 1D or 2D boolean
+        numpy.ndarry (depending on the dimension of `T`) or a function that can be
+        applied to each time series in `T`. Alternatively, for  maximum flexibility, a
+        list (with length equal to the total number of time series) may also be used.
+        In this case, T_subseq_isconstant[i] corresponds to the i-th time series T[i]
+        and each element in the list can either be 1D boolean np.ndarray, a function,
+        or None.
+
+    T_subseq_isfinite : numpy.ndarray, default None
```

Comment:
I added this param since `fix_isconstant_isfinite_conflicts` , which is used in this function, has the param `T_subseq_isfinite`. However, I now think it should bettter be removed as this may lead to a strange result if a user provides their own `T_subseq_isfinite` that is different than the outcome of `core.rolling_isfinite(T, m)`.

Reply:
okay. So, let's keep the parameter `T_subseq_isfinite` for now. If we assume this is going to be used by advanced users, they probably won't mess it up.